packages/ember-metal/lib/watch_key.js

```diff
@@ -79,9 +77,6 @@ if (MANDATORY_SETTER) {
 }
 
 export function unwatchKey(obj, keyName, _meta) {
-  if (typeof obj !== 'object' || obj === null) {
```

Comment:
since `peekMeta` has assertion for non-object / non-function values, it is better to assert than be silent to uncover unnecessary calls of unwatchKey

Reply:
Yes, I agree, though we should check that this doesn't cause any crashes in a few real world apps.

Reply:
will do

Reply:
checked in both emberobserver and emberaddons, both working properly and no regression in benchmarks https://cl.ly/441V333w3R3R/results.pdf